### PR TITLE
Async all the things

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "trend-detector",
   "version": "0.0.1",
   "description": "Function that watches Orders and records ranked quantities for Products in Redis",
-  "main": "func.js",
+  "main": "main.js",
```
Comment (scothis) [inline package.json]: Revert if you want to run as a riff function

File: main.js [hunk 2]
```diff
@@ -1,14 +1,22 @@
-const myFunction  = require('./func');
+const myFunction = require('./func');
```
Comment (scothis) [inline main.js]: I still don’t understand why the main.js module exists.  It looks like a test, but isn’t hooked into the test script or have any assertions.

File: func.js [hunk 3]
```diff
@@ -1,26 +1,40 @@
...
-    for (const productId of Object.keys(order.products)) {
-        client.zadd("top-orders", "incr", order.products[productId], productId);
-    }
-    client.zrevrange("top-orders", 0, -1, "WITHSCORES", (err, members) => {
-        for (let i = 0; i <= members.length - 1; i += 2) {
-            console.log(`Item ${members[i]}, sold ${members[i + 1]} time(s)`)
-        }
-    });
+    return zaddAll(order.products)
+        .then(_ => zrevrangeAsync("top-orders", 0, -1, "WITHSCORES"))
```
Comment (scothis) [inline func.js]:
```suggestion
        .then(() => zrevrangeAsync("top-orders", 0, -1, "WITHSCORES"))
```

File: func.js [hunk 4]
```diff
@@ -1,26 +1,40 @@
...
 
 module.exports.$destroy = () => {
-    client.quit();
+    return quitAsync();
 };
+
+function zaddAll(products) {
+    return Promise.all(Object.keys(products)
+        .map(key => [key, products[key]])
+        .map(tuple => zaddAsync("top-orders", "incr", tuple[1], tuple[0])));
```
Comment (scothis) [inline func.js]: the tuple doesn’t seem to add much value

File: main.js (hunk 2 above)
Comment (fbiville) [reply main.js]: it's a manual test (as documented in the README). That's how I test it locally. I can spend more time and turn that into a real test if you do not like the current form.

**Comment by fbiville** [comment]
Merged as b7a25efe761e93c90e9689f8f3af3d4effe892a5

**Pull request closed without merging**